compiler/rustc_ty_utils/src/assoc.rs

```diff
@@ -258,7 +258,9 @@ fn associated_type_for_impl_trait_in_trait(
     let local_def_id = trait_assoc_ty.def_id();
     let def_id = local_def_id.to_def_id();
 
-    trait_assoc_ty.feed_hir();
+    // Set the parent HIR for this item to be the `Node::Ty` of the `impl Trait` in trait.
+    let parent_ty = tcx.parent_hir_id(tcx.local_def_id_to_hir_id(opaque_ty_def_id));
```

Comment:
This HIR id must live between `local_def_id` and `trait_def_id`, otherwise there's a mismatch between `DefId` and `HirId` parenting trees.

But that's not the case in this PR, if the comment is correct then `parent_ty` lives inside `fn_def_id`, which is a neighbor of `local_def_id`.

Reply:
The drawback of https://github.com/rust-lang/rust/pull/123415 is that it assigns the HIR parent silently and we won't notice if some parent is actually missing.

Luckily, both associated opaque types and implicit statics, are directly parented by owner nodes (traits, impls, or statics), so the default zero owner-local id is correct in all these case, and no mismatch with `DefId` hierarchy happens.